numba/core/runtime/nrt.cpp

```diff
@@ -348,35 +298,35 @@ void NRT_MemInfo_call_dtor(NRT_MemInfo *mi) {
     NRT_MemInfo_destroy(mi);
 }
 
-void NRT_MemInfo_release(NRT_MemInfo *mi) {
+extern "C" void NRT_MemInfo_release(NRT_MemInfo *mi) {
     NRT_Debug(nrt_debug_print("NRT_MemInfo_release %p refct=%zu\n", mi,
-                                                            mi->refct));
+                                                            mi->refct.load()));
     assert (mi->refct > 0 && "RefCt cannot be 0");
     /* RefCt drop to zero */
-    if (TheMSys.atomic_dec(&mi->refct) == 0) {
+    if ((--mi->refct) == 0) {
```

Comment:
I always find this hard to evaluate the order of operations (pre-decrement with dereference) - is it worth making this
```suggestion
    if ((--(mi->refct)) == 0) {
```

for people with similar struggles, or is it clear to most?

Reply:
Agree, fixed in 6a2389e.